File: src/main/java/dkoryakin/aston/demo/api/body/request/AccountTransferPostBody.java
package dkoryakin.aston.demo.api.body.request;

import dkoryakin.aston.demo.infrastructure.validator.PinCodeConstraint;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AccountTransferPostBody {

    @PinCodeConstraint
    private String pin;

    @NotNull
    private BigDecimal amount;

    @NotNull
    private Long accountId;
}

File: src/main/java/dkoryakin/aston/demo/api/body/AccountTransferPostBody.kt
package dkoryakin.aston.demo.api.body

import dkoryakin.aston.demo.infrastructure.validator.PinCodeConstraint
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import org.hibernate.validator.constraints.Length


data class AccountTransferPostBody(@PinCodeConstraint val pin: String,
                                   @NotNull val amount:Double,
                                    @NotNull val accountId: Long
) { }